src/Hazelcast.Net/Protocol/RemoteException.cs

```diff
@@ -126,8 +154,19 @@ public RemoteException(RemoteError error, string message, Exception innerExcepti
         private RemoteException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
-            Error = (RemoteError) info.GetInt32("error");
-            Retryable = info.GetBoolean("retryable");
+            MemberId = info.GetGuid(nameof(MemberId));
+            Error = (RemoteError) info.GetInt32(nameof(Error));
+            Retryable = info.GetBoolean(nameof(Retryable));
+            ServerStackTrace = info.GetString(nameof(ServerStackTrace));
+        }
+
+        /// <summary>
+        /// Gets the unique identifier of the member which threw the exception.
+        /// </summary>
+        public Guid MemberId
+        {
+            get => _memberId;
+            set => Data[nameof(MemberId)] = _memberId = value;
```

Comment:
Why to use both dictionary and private field?

Reply:
totally, pushing a simplification